### PR TITLE
Fix hot module reload

### PR DESCRIPTION
**Pull request by freddi301**
## Short description

Now react edits do not cause whole page reload (boost productivity)

## How to test
Try to edit some jsx and see it no longer causes full page reload

### CODE REVIEW

File: src/navigation/RouterConfig.tsx [hunk 1]
```diff
@@ -63,7 +63,9 @@ export const RouterConfig = ({ userType }: { user: any; userType: string }) => {
           break;
         case AgreementState.PendingAgreement:
         case AgreementState.ApprovedAgreement:
-          history.push(DASHBOARD);
+          if (location.pathname === "/") {
```
Comment (Hantex9) [inline src/navigation/RouterConfig.tsx:66]: What is the behavior of that condition, why is it needed?

**Review by Hantex9 — COMMENTED** [review]
Great job, the hot loader works like a charm and it doesn't have any library trouble 🚀

BTW I'm leaving just a comment on a snippet of code just to clarify your choice.

File: src/navigation/RouterConfig.tsx (hunk 1 above)
Comment (freddi301) [reply src/navigation/RouterConfig.tsx:66]: This Effect triggers at page load. The previous behaviour was that if the user reloads the page, react router correctly renders the same route, but this effect redirects always on reload to dashboard route, i think this behaviour is accidental. This condition now redirects to dashboard only if we are the root route of the domain

Comment (freddi301) [reply src/navigation/RouterConfig.tsx:66]: this effect prevent hor reloading to work properly also

**Review by Hantex9 — APPROVED** [review]
LGTM!

**Pull request merged**